sky/packages/sky/lib/widgets/date_picker.dart

```diff
@@ -0,0 +1,392 @@
+    firstDate = source.firstDate;
+    lastDate = source.lastDate;
+  }
+
+  DatePickerMode _mode = DatePickerMode.day;
+
+  void _handleModeChanged(DatePickerMode mode) {
+    setState(() {
+      _mode = mode;
+    });
+  }
+
+  void _handleYearChanged(DateTime dateTime) {
+    setState(() {
+      _mode = DatePickerMode.day;
+    });
+    if (onChanged != null)
+      onChanged(dateTime);
+  }
+
+  static const double calendarHeight = 210.0;
+
+  Widget build() {
+    Widget header = new DatePickerHeader(
+      selectedDate: selectedDate,
+      mode: _mode,
+      onModeChanged: _handleModeChanged
+    );
+    Widget picker;
+    switch (_mode) {
+      case DatePickerMode.day:
+        picker = new MonthPicker(
+          selectedDate: selectedDate,
+          onChanged: onChanged,
+          firstDate: firstDate,
+          lastDate: lastDate,
+          itemExtent: calendarHeight
+        );
+        break;
+      case DatePickerMode.year:
+        picker = new YearPicker(
+          selectedDate: selectedDate,
+          onChanged: _handleYearChanged,
+          firstDate: firstDate,
+          lastDate: lastDate
+        );
+        break;
+    }
+    return new BlockBody([header, new Container(height: calendarHeight, child: picker)]);
+  }
+
+}
+
+// Shows the selected date in large font and toggles between year and day mode
+class DatePickerHeader extends Component {
+  DatePickerHeader({ this.selectedDate, this.mode, this.onModeChanged }) {
+    assert(selectedDate != null);
+    assert(mode != null);
+  }
+
+  DateTime selectedDate;
+  DatePickerMode mode;
+  DatePickerModeChanged onModeChanged;
+
+  EventDisposition _handleChangeMode(DatePickerMode value) {
+    if (value == mode)
+      return EventDisposition.ignored;
+    onModeChanged(value);
+    return EventDisposition.processed;
+  }
+
+  Widget build() {
+    ThemeData theme = Theme.of(this);
+    typography.TextTheme headerTheme;
+    Color dayColor;
+    Color yearColor;
+    switch(theme.primaryColorBrightness) {
+      case ThemeBrightness.light:
+        headerTheme = typography.black;
+        dayColor = mode == DatePickerMode.day ? colors.black87 : colors.black54;
+        yearColor = mode == DatePickerMode.year ? colors.black87 : colors.black54;
+        break;
+      case ThemeBrightness.dark:
+        headerTheme = typography.white;
+        dayColor = mode == DatePickerMode.day ? colors.white87 : colors.white54;
+        yearColor = mode == DatePickerMode.year ? colors.white87 : colors.white54;
+        break;
+    }
+    TextStyle dayStyle = headerTheme.display3.copyWith(color: dayColor, height: 1.0, fontSize: 100.0);
+    TextStyle monthStyle = headerTheme.headline.copyWith(color: dayColor, height: 1.0);
+    TextStyle yearStyle = headerTheme.headline.copyWith(color: yearColor, height: 1.0);
+    DateTime firstDate = new DateTime(1900);
+    DateTime lastDate = new DateTime(2101);
+
+    return new Container(
+      child: new BlockBody([
```

Comment:
Doesn't work, same problem as above. Will change to Column once I figure out why Column isn't working